### PR TITLE
Adding a FAQ for the project

### PR DESCRIPTION
**Pull request by DavidDeSimone**
I've gotten these questions through various mediums, sometimes multiple times. Seems like a good point to add an FAQ.

### CODE REVIEW

File: faq.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+## How should I use JS/TS as an existing package maintainer?
+
+If you have a large elisp package, our guidance is not that you should rewrite your entire package in JS/TS. Instead, we encourage package maintainers to explore using JS/TS's Async I/O and Threading capabilities to improve performance their hot code paths on emacs-ng. Using `(featurep 'emacs-ng)`, you can include an import statement for a JS/TS package that defun's functions for you to use. Our [Getting Started Guide](https://github.com/emacs-ng/emacs-ng/blob/master/getting-started.md) is a good place to start.
+
+## How does adding JS/TS affect your ability to merge future emacs improvements?
+
+JS/TS is almost a completely additive layer, we have made extremely minimal C changes. As of writing this, we have only made a small edit to a single line of C. WebRender, while still in development, has also made minimal C changes. We have the ability to cleanly merge upstream patches, including compatibility with the `native-comp` branch of emacs.
```
Comment (brotzeit) [inline faq.md:13]: I don't know if "compatibility" is the right term here. It's part merged and of the project.

Comment (brotzeit) [reply faq.md:13]: Maybe we should mention that we do upstream merges by merging the native-comp branch(it's still not merged right ?).

Comment (DavidDeSimone) [reply faq.md]: Updated, let me know if you like this wording better.

Comment (brotzeit) [reply faq.md]: Perfect.

**Review by brotzeit — APPROVED** [review]

**Pull request merged**